test/e2e/scheduling/preemption.go

```diff
@@ -737,3 +736,21 @@ func waitForPreemptingWithTimeout(f *framework.Framework, pod *v1.Pod, timeout t
 	})
 	framework.ExpectNoError(err, "pod %v/%v failed to preempt other pods", pod.Namespace, pod.Name)
 }
+
+func patchNode(client clientset.Interface, old *v1.Node, new *v1.Node) error {
```

Comment:
I think this is an improvement, but I think the error described in the issue may be misleading as I am also seeing this: `Jun  9 12:22:18.370: failed pod observation expectations: rs1 had more than 2 pods created: 3`

Reply:
Are you referring to another test failure (other than `basic preemption works`)? That may be related (but I need to confirm) - the node is not updated successfully with proper "fake resource", hence the rs doesn't get expected replicas scheduled. But anyway, let focus on the `basic preemption works` failure in this PR.

Reply:
Ah yes, this new failure appears to be different. I will open a separate issue :+1:

Reply:
Please, thanks - I'm aware of that one: "PreemptionExecutionPath runs ReplicaSets ". Hope this one fixes that one as well.